pkg/clusteragent/admission/common/label_selectors.go

```diff
@@ -13,64 +13,75 @@ import (
 	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
 )
 
-// DefaultLabelSelectors returns the mutating webhooks object selector based on the configuration
-func DefaultLabelSelectors(useNamespaceSelector bool) (namespaceSelector, objectSelector *metav1.LabelSelector) {
-	var labelSelector metav1.LabelSelector
+// LabelSelectorsConfig configures the namespace label selectors for admission webhooks
+type LabelSelectorsConfig struct {
+	// ExcludeNamespaces lists namespaces to exclude from webhook invocation
+	ExcludeNamespaces []string
 
-	if pkgconfigsetup.Datadog().GetBool("admission_controller.mutate_unlabelled") ||
-		pkgconfigsetup.Datadog().GetBool("apm_config.instrumentation.enabled") ||
-		len(pkgconfigsetup.Datadog().GetStringSlice("apm_config.instrumentation.enabled_namespaces")) > 0 {
-		// Accept all, ignore pods if they're explicitly filtered-out
-		labelSelector = metav1.LabelSelector{
-			MatchExpressions: []metav1.LabelSelectorRequirement{
-				{
-					Key:      EnabledLabelKey,
-					Operator: metav1.LabelSelectorOpNotIn,
-					Values:   []string{"false"},
-				},
-			},
-		}
+	// IncludeNamespaces lists namespaces to explicitly include (mutually exclusive with ExcludeNamespaces)
+	// If set, only these namespaces will be targeted
+	IncludeNamespaces []string
+}
+
+// DefaultLabelSelectors returns namespace and object selectors for admission webhooks.
+// A nil object selector is returned if useNamespaceSelector is true.
+func DefaultLabelSelectors(useNamespaceSelector bool, config LabelSelectorsConfig) (*metav1.LabelSelector, *metav1.LabelSelector) {
+	nsSelector := &metav1.LabelSelector{}
+	var objSelector *metav1.LabelSelector
+	if !useNamespaceSelector {
+		objSelector = &metav1.LabelSelector{}
+		applyAdmissionEnabledSelectors(objSelector)
 	} else {
-		// Ignore all, accept pods if they're explicitly allowed
-		labelSelector = metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				EnabledLabelKey: "true",
-			},
-		}
+		applyAdmissionEnabledSelectors(nsSelector)
 	}
 
+	applySelectorConfig(nsSelector, config)
```

Comment:
One thing to consider here is that this adds namespace selectors even when the parameter `useNamespaceSelector` is false.

I think using namespace selectors here is the correct option, but the naming is confusing, so we could improve it or add a comment explaining this.

I think `useNamespaceSelector` is only used in very specific cases. Only in kubernetes versions between 1.10 and 1.14 and there's an agent option that needs to be enabled `admission_controller.namespace_selector_fallback`: https://github.com/DataDog/datadog-agent/blob/439b1de10795f503c0de7c0bbf17375c836ce667/pkg/clusteragent/admission/util.go#L42